app/controllers/supervisor/base_controller.rb

```diff
@@ -0,0 +1,18 @@
+class Supervisor::BaseController < ApplicationController
+
+  before_action :must_be_supervisor
+  helper_method :supervised_teams
+
+  protected
+
+  def must_be_supervisor
+    unless signed_in? && current_user.roles.includes?('supervisor')
```

Comment:
It would be nicer if the second part of this condition is moved to a method (the roles part). That way you could write this instead:

    unless signed_in? && current_user.supervisor?

The corresponding method could be as simple as:

```ruby
class User < ActiveRecord::Base
  def supervisor?
    roles.includes?('supervisor')
  end
end
```